UI/Shell-UI: Migrate to Typescript

ChengYanJin opened the pull request:
**Component**: UI, shell-ui


**Context**: 

**Summary**:
Migrate to Typescript
Remove all the flow typing
Bump the version of `webpack.dev.server`

**Acceptance criteria**:

ChengYanJin committed:
ui: run the script to migrate to typescript

ChengYanJin committed:
ui: remove flow typing

ChengYanJin committed:
ui: fix tests following the migration to typescript

bert-e commented:
# Hello chengyanjin,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/metalk8s/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @chengyanjin

* @teddyandrieux

* @lucieleonard

* @gdemonet

* @thomasdanan

* @jbwatenbergscality

ChengYanJin committed:
shell-ui: Fix the test

ChengYanJin committed:
shell-ui: remove the flow

ChengYanJin committed:
ui/shell-ui: Bump webpack.dev.server and fix breaking change

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @gdemonet

* @lucieleonard

* @jbwatenbergscality

* @teddyandrieux

* @chengyanjin

ChengYanJin committed:
shell-ui: increase the default asset size limit defined in webpack.common.js

ChengYanJin committed:
tool/crd-client-generator: generate the client base on metalk8s volume crd

ChengYanJin committed:
shell-ui: remove the unused dependencies ( react-to-webcomponent & quicklink & prop-types )

ChengYanJin committed:
ui: Fix the build error - can't resolve 'querystring'

Module not found: Error: Can't resolve 'querystring'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "querystring": require.resolve("querystring-es3") }'
	- install 'querystring-es3'

JBWatenbergScality commented on shell-ui/webpack.common.js:
```diff
@@ -59,23 +59,34 @@ module.exports = {
+  },
+  performance: {
+    hints: 'warning',
+    // ~732 KiB for production
+    // ~10.5 MiB for development because flow increase the size of assets.
+    maxAssetSize: process.env.NODE_ENV === 'production' ? 750000 : 11000000,
```
nit: Probably those comments/configs are not accurate

JBWatenbergScality commented on tools/crd-client-generator-js/README.md:
```diff
@@ -2,7 +2,7 @@
 
 It generates a flow client based on `@kubernetes/client` custom objects client from a Kubernetes `CustomResourceDefinition`.
```
```suggestion
It generates a typescript client based on `@kubernetes/client` custom objects client from a Kubernetes `CustomResourceDefinition`.
```

JBWatenbergScality reviewed (approved):
LGTM!

ChengYanJin commented:
/approve

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/66a91a80b8ff1e4a4d7c716dfd0a2907fd40ae1c) did not succeed in branch improvement/migrate-to-typescript.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/125.0`



The following branches will **NOT** be impacted:

* `development/123.0`
* `development/124.0`
* `development/124.1`
* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.11`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/125.0`



The following branches have **NOT** changed:

* `development/123.0`
* `development/124.0`
* `development/124.1`
* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.11`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



Please check the status of the associated issue None.

Goodbye chengyanjin.
